### PR TITLE
carli/1987_changes_to_tsv_files

### LINKED ISSUE #1987: User requested changes to tsv files
During a recent CARTA webinar we gave at NRAO, a user (Peter Teuben from Maryland) asked whether CARTA output text files that could be used by other software (to plot or do statics). I sent him a .tsv file from a histogram and from spectral fitting. He had the following comments:

- the default name for the tsv file was named something like "<filename>-Z-profile-Region_1-Statistic_FluxDensity-Cooridnate_Current-2022-08-25-16-50-14.tsv, which has a typo (Cooridnate instead of Coordinate)

- don't allow spaces in filenames (the files I sent him did not have spaces in their names, so maybe this is already done)

- Make sure there is not a blank line at the end, which he has seen programs/compilers fail on

### PR DESCRIPTION
**Pull request by crocka**
**Description**
The ``exportTsvFile`` function in ``export.ts`` is where the exported tsv files are named. ``imageName`` and ``plotName`` are parameters of the function that go into the names of the exported tsv files. Checked according to user's comment, no space is observed in ``imageName`` or ``plotName``, and no ``\n`` is found, so only typos are corrected. This PR resolves #1987.

**Checklist**

For linked issues (if there are):
- [x] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] ~e2e test passing~ / corresponding fix added
- [x] changelog updated / ~~no changelog update needed~~
- [x] ~~protobuf updated to the latest dev commit~~ / no protobuf update needed
- [x] `BackendService` unchanged / ~~`BackendService` changed and corresponding ICD test fix added~~

### CODE REVIEW

**Comment by kswang1029** [comment]
@crocka I tested with an image file named like "spire500 ext.fits" (there is a space in the filename) then the exported tsv filename would have the space too. I suggest we replace space with __ (double-underscore).

Attached is the example file exported with v4-beta. It appears that there is a blank line in the end.
[spire500 ext.fits.HDU_1_image-X-profile-2023-05-23-14-52-49.tsv.zip](https://github.com/CARTAvis/carta-frontend/files/11540481/spire500.ext.fits.HDU_1_image-X-profile-2023-05-23-14-52-49.tsv.zip)

**Comment by kswang1029** [comment]
tsv and txt files are now with "space" in filename replaced by "__". 👍 The \n issue is also fixed.
One last question, do we want to apply the space -> __ treatment to png filename too for consistency?

File: src/utilities/export/export.ts [hunk 1]
```diff
@@ -10,26 +10,26 @@ export function getUnixTimestamp() {
...
 export function exportTsvFile(imageName: string, plotName: string, content: string) {
-    const tsvData = `data:text/tab-separated-values;charset=utf-8,${content}\n`;
+    const tsvData = `data:text/tab-separated-values;charset=utf-8,${content}\n`.trim();
     const dataURL = encodeURI(tsvData).replace(/#/g, "%23");
 
     const a = document.createElement("a") as HTMLAnchorElement;
     a.href = dataURL;
 
-    a.download = `${imageName}-${plotName.replace(" ", "-")}-${getTimestamp()}.tsv`;
+    a.download = `${imageName.replace(" ", "__")}-${plotName.replace(" ", "-")}-${getTimestamp()}.tsv`;
```
Comment (YuHsuan-Hwang) [inline src/utilities/export/export.ts]: This will only replace the first space. Perhaps use `replaceAll`. Same for the other function.

**Review by YuHsuan-Hwang — CHANGES_REQUESTED** [review]
Text filename in `ProfileFittingComponent` need to be changed to `Profile_Fitting_Result_Log`. The space is modified to double underscore.

**Review by YuHsuan-Hwang — APPROVED** [review]

**Review by kswang1029 — APPROVED** [review]

**Comment by YuHsuan-Hwang** [comment]
This is ready for merge. @crocka Please update the change log, thanks.

**Pull request merged**